orchagent/orchdaemon.cpp

```diff
@@ -654,3 +663,50 @@ bool OrchDaemon::warmRestartCheck()
     gSwitchOrch->restartCheckReply(op,  data, values);
     return ret;
 }
+
+void
+OrchDaemon::check_and_exit()
+{
+    SWSS_LOG_ENTER();
+
+    swss::SubscriberStateTable tbl(m_configDb, "CONTAINER_FEATURE");
+    std::deque<KeyOpFieldsValuesTuple> entries;
+
+    tbl.pops(entries);
+
+    for (auto& entry: entries) {
+        string key = kfvKey(entry);
+
+        if (key == "swss") {
+            auto values = kfvFieldsValues(entry);
+
+            for (auto it : values)
+            {
+                string field = fvField(it);
+                string value = fvValue(it);
+
+                if (field == "orchagent_exit_on_switch_shutdown") {
```

Comment:
same for "orchagent_exit_on_switch_shutdown" and "swss" if you are expecting those to be received, then someone had to send those, so you could put those constants in some header and then reuse same header in sender?

Reply:
Yes/No. Need your help to decide.

This is *only* a temp knob, more for an easy way to revert back to legacy behavior, if the freeze-mode was found to be impacting or causing regression. If not for this knob, we will be forced to build a new image.

Said that,  we either have to record this "orchagent_exit_on_switch_shutdown" and "swss" or remove this check. Being in transient phase, would it be OK to raise a github issue, so we get to it, in sometime ?

Or should we record it in code in some form ?

Can you please suggest?